docs/docs/auth.md

```diff
@@ -14,6 +14,13 @@ import org.http4s.dsl.io._
 import org.http4s.server._
 ```
 
+We will also require the `Ykind-projector` scalac flag, which you can add into your `build.sbt` using
+```scala mdoc:silent
+scalacOptions ++= Seq(
+  "-Ykind-projector"
+)
+```
```

Comment:
Thanks for this! We have to take into account a couple considerations.
1. This scalac flag only works for Scala 3. For Scala 2 a compiler plugin is required.
  https://github.com/typelevel/kind-projector
2. Actually, this is blanket advice that may be required for many examples, not just auth. So I wonder if we should put it in one place, and add a reminder for every page? Not sure.

Reply:
Yes that's a great point - I simply didn't know this. I can't find a section in the docs that quite fits this, maybe I should create a new section, after "Quick Start", and call it something like "General setup"? And that can combine this type of info? Maybe it's only this extension, but I'd guess there's more to come.

Reply:
It's required for any page that makes use of the `*` syntax as a type parameter (a lot of them, if not most of them, AFAICT).

BTW, maybe we can add this information to the quick start page. That page mentions the partial unification compiler flag for example.
https://http4s.org/v0.23/docs/quickstart.html

That page also points to the g8 template for http4s, which actually sets this up:
https://github.com/http4s/http4s.g8/blob/e5b55668c29836201e4a554ee1f338a897dfff22/src/main/g8/build.sbt#L26